server/service/src/stock_line/update.rs

```diff
@@ -168,6 +193,45 @@ fn generate_location_movement(
     Ok(movement)
 }
 
+fn generate_barcode_row(
+    connection: &StorageConnection,
+    existing: StockLineRow,
+    value: String,
+) -> Option<BarcodeRow> {
+    // for an empty string, simply unlink the barcode
+    if value.is_empty() {
+        return None;
+    }
+
+    let filter = BarcodeFilter::new()
+        .item_id(EqualFilter::equal_to(&existing.item_id))
+        .pack_size(EqualFilter::equal_to_i32(existing.pack_size));
```

Comment:
So does this mean we consider ALL barcode to be unique by item id and pack size only, and we don't expect any barcodes that are by batch ? 

Like if you had two different barcodes for the same item with the same pack size but different batch, and you assigned one barcode to one batch then assigned second barcode to second batch it would overwrite first barcode ?

Reply:
the barcode bit is constant - the batch and expiry are separate parts, they are after `(10)` and `(17)` with the 'barcode' part being before that. It means that we can save that 'barcode' bit against an item and pack size combination in order to match items when you scan.

Reply:
@mark-prins @andreievg If I understand your conversation correctly, the term you're needing for the 'barcode' part is GTIN (Global Trade Identification Number), the (01) element of the barcode. The GTIN varies with pack size (same item with different pack sizes = different GTINs) but not batch or expiry (same item with same pack size but different batches and expiry dates = same GTIN). Hope some of that's helpful!

Reply:
ahh oki, this sounds like we need to really rename `value` and other references to `value` of the barcode (sometimes we call the value of the barcode, just barcode) to GTIN (to avoid further confusion) ? 

Thanks @GaryWilletts, but I am still very confused, I wondering if there is a simplified summary of our understanding of the spec somewhere (i.e. not sure if we want devs to invest the time to[ study the full spec](https://www.gs1us.org/content/dam/gs1us/documents/industries-insights/standards/Guideline-An-Introduction-to-the-Global-Trade-Item-Number-GTIN.pdf)), I did look at [our wiki](https://wiki.sussol.net/doku.php/msupply:design:special:barcoding?s[]=gtin) but it's more about how we implement it rather then a summary of what it is

Reply:
thanks @GaryWilletts - that's exactly it. And renaming to value to GTIN would be sensible

Reply:
@andreievg reading the spec is a horrible experience(!) and really not worth it, I've written a summary on that wiki page. Let me know if you need more added to it, I'd like to save you the pain 😉

Reply:
> And renaming to value to GTIN would be sensible

https://github.com/openmsupply/open-msupply/issues/1767